dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/TemplatePage.java

```diff
@@ -138,37 +135,23 @@ public String init() {
                 template.setTermsOfUseAndAccess(terms);
             }
 
-            updateDatasetFieldsIncludeFlag();
         } else if (ownerId != null) {
-            // create mode for a new template
 
             editMode = TemplatePage.EditMode.CREATE;
             template = new Template(this.dataverse);
             TermsOfUseAndAccess terms = new TermsOfUseAndAccess();
             terms.setTemplate(template);
             terms.setLicense(TermsOfUseAndAccess.License.CC0);
             template.setTermsOfUseAndAccess(terms);
-            updateDatasetFieldsIncludeFlag();
+
+            List<DatasetField> datasetFields = datasetFieldsInitializer.prepareDatasetFieldsForEdit(template.getDatasetFields(), dataverse);
```

Comment:
There is a difference between dataverse connected to template and metadataBlocksDataverse in this function. Passed dataverse must be a dataverse defining what dataset fields are available

Reply:
The dataverse that is passed here is the dataverse that defines which dataset fields are available. If Root (1 mdb) -> testDataverse (3 mdb) then if template is created in testDataverse, 3 mdbs are available for Template.

Reply:
Yes, because testDataverse is the root for metadata blocks.
What will happen if testDataverse will have flag set on - inherit metadata blocks from root?